### PR TITLE
Add metrics support, and add operation_count metric

### PR DESCRIPTION
**Pull request by bswartz**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add metrics support, and add operation_count metric

**Special notes for your reviewer**:
While adding metrics support, also synchronize some other code with CSI controllers.

**Does this PR introduce a user-facing change?**:
```release-note
The controller now exposes metrics covering the counts and results of validation operations.
```

### CODE REVIEW

File: pkg/metrics/metrics_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+	expected :=
+		`# HELP process_start_time_seconds [ALPHA] Start time of the process since unix epoch in seconds.
+        # TYPE process_start_time_seconds gauge
+        process_start_time_seconds 0
+# HELP volume_data_source_validator_volume_data_source_validator_operation_count [ALPHA] Number of validations operations by result
+# TYPE volume_data_source_validator_volume_data_source_validator_operation_count counter
+volume_data_source_validator_operation_count{result="test_result_1"} 1
+volume_data_source_validator_operation_count{result="test_result_2"} 2
+`
```
Comment (xing-yang) [inline pkg/metrics/metrics_test.go:93]: So we don't have a metric about how long the operation takes, i.e., operation_total_seconds?

**Comment by xing-yang** [comment]
Is CI not running?  I don't see anything here: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_volume-data-source-validator/14/pull-kubernetes-csi-volume-data-source-validator/1454163767536390144

Unit tests should be running with CI.  I don't see it here.

**Comment by xing-yang** [comment]
volume-data-source-validator should have a unit test job like this one:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-csi/external-snapshotter/external-snapshotter-config.yaml#L345

**Comment by xing-yang** [comment]
Since unit tests are not shown in CI, can you post the test results here?

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (bswartz) [reply pkg/metrics/metrics.go:2]: The code was written last year though, in 2021

File: pkg/metrics/metrics_test.go (hunk 1 above)
Comment (bswartz) [reply pkg/metrics/metrics_test.go:93]: The validator is just a webhook that runs are admission time. It has no idea what happens after that.

Information about how long population takes will be supplied by the populators themselves -- and support for that is going into the library. I started work on that last year then got side tracked, but I have to get it done for 1.24.

**Comment by bswartz** [comment]
Test results:

$ go test ./...
?   	github.com/kubernetes-csi/volume-data-source-validator/cmd/volume-data-source-validator	[no test files]
ok  	github.com/kubernetes-csi/volume-data-source-validator/pkg/data-source-validator	0.109s
ok  	github.com/kubernetes-csi/volume-data-source-validator/pkg/metrics	0.046s
?   	github.com/kubernetes-csi/volume-data-source-validator/release-tools	[no test files]

**Pull request merged**